inmaputil/download.go

```diff
@@ -0,0 +1,105 @@
+package inmaputil
+
+import (
+	"github.com/mholt/archiver"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/cavaliercoder/grab"
+)
+
+func Download(path string) string {
+	// Check if local file exists
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		return path;
+	} 
+
+	// Check if path doesn't start with http or https
+	if !strings.HasPrefix(path, "http://") && !strings.HasPrefix(path, "https://") {
+		return path;
+	}
+
+	// Prepare temporary directory
+	if err := os.MkdirAll("./temp/", os.ModePerm); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed creating temporary folder to download file. %v\n", err)
```

Comment:
panic(fmt.Errorf("inmaputil: failed creating temporary download directory: %v", err)

Reply:
We want to use a panic here because it's something that would happen because of a system error rather than a user error.